app/GraphQL/Mutations/Login.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace App\GraphQL\Mutations;
+
+use App\Models\User;
+use GraphQL\Error\Error;
+use Illuminate\Support\Facades\Auth;
+
+class Login
+{
+    /**
+     * @param  null  $_
+     * @param  array<string, mixed>  $args
+     */
+    public function __invoke($_, array $args): User
+    {
+        $guard = Auth::guard();
```

Comment:
Should be `$guard = Auth::guard(config('sanctum.guard', 'web'));`

We need to force the use of sanctum's default guard here, because `web` might not always be the project's default session-based guard.

Also, if we get a request with a token instead of a valid cookie, the token-based auth driver would not allow us to log the user in (as it doesn't know the concept of sessions, that's the job of the 'session' driver → duh).

Reply:
In my example, I'm returning the user straight away if it's already attached to the session, before attempting sign-in:

```
if ($guard->user()) {
    return $guard->user();
}
```

Reply:
> Should be $guard = Auth::guard(config('sanctum.guard', 'web'));

I changed that bit because I could not find a setting `guard` within `config/sanctum.php`, is it hidden within the default settings?

What you are saying makes sense.

Reply:
> I'm returning the user straight away if it's already attached to the session, before attempting sign-in

What if an already authenticated user sends a different set of credentials to authenticate as another user? It seems counter-intuitive that their login will not change and the attempt will be silently discarded.

Is there a reason where the behaviour you proposed would be advantaguous?

Reply:
> I changed that bit because I could not find a setting guard within config/sanctum.php, is it hidden within the default settings?

It's definitely read from by Sanctum, they just don't expose it to the config file. I guess that's by mistake, why else would you read from the user config  ¯\\_(ツ)_/¯
https://github.com/laravel/sanctum/blob/2.x/src/Guard.php#L54

Anyways, the guard must use a session driver.

> What if an already authenticated user sends a different set of credentials to authenticate as another user

Good point. Switching sessions right away might be unexpected behaviour, might even make sense to throw here.